gcp_variant_transforms/transforms/merge_conflicts.py

```diff
@@ -0,0 +1,127 @@
+  _MAX_FILE_NAME_NUM = 5
+
+  def __init__(self, definition, file_name):
+    # type: (OrderedDict[str, Union[str, int]], str) -> None
+    """Initialized a _ConflictDetails object.
+
+    Creates a dictionary that maps field definition to a list of file names.
+    Upon initialization, add ``definition`` and ``file_name`` to the dictionary.
+    """
+    self.conflicts = {self._format_definition(definition): list([file_name])}
+
+  def _add_details(self, other):
+    # type: (_ConflictDetails) -> None
+    for conflict, file_names in other.conflicts.iteritems():
+      if conflict not in self.conflicts.keys():
+        self.conflicts[conflict] = file_names
+        continue
+
+      for file_name in file_names:
+        if len(self.conflicts[conflict]) == self._MAX_FILE_NAME_NUM:
+          break
+        self.conflicts[conflict].append(file_name)
+
+  def _format_definition(self, definition):
+    # type: (OrderedDict[str, Union[str, int]]) -> str
+    """Formats the conflict string given one field definition.
+
+    Only num and type is considered for the conflicts, and the definition forms
+    a string in the format of `num={value} type={value}`.
+    """
+    conflict_keys = [VcfParserConstants.NUM, VcfParserConstants.TYPE]
+    formatted_conflict = []
+    for key in conflict_keys:
+      formatted_conflict.append(key + '=' + str(definition[key]))
+    return ' '.join(formatted_conflict)
+
+
+class VcfHeaderConflicts(object):
+  """Container for header conflicts."""
+
+  def __init__(self, vcf_header=VcfHeader()):
+    # type: (VcfHeader) -> None
+    """Initializes a VcfHeaderConflicts object.
+
+    Creates two dictionaries that map info and format field id/key to
+    _ConflictDetails respectively.
+    """
+
+    file_name = vcf_header.file_name
+    self.info_conflicts = {}
+    self.format_conflicts = {}
+    for key, val in vcf_header.infos.iteritems():
+      self.info_conflicts[key] = _ConflictDetails(val, file_name)
+    for key, val in vcf_header.formats.iteritems():
+      self.format_conflicts[key] = _ConflictDetails(val, file_name)
+
+
+class _ConflictsMerger(object):
+  """Class for merging two :class:`VcfHeaderConflicts`s."""
+
+  def merge(self, first, second):
+    # type: (VcfHeaderConflicts, VcfHeaderConflicts) -> None
+    """Updates ``first``'s conflicts with values from ``second``."""
+    if (not isinstance(first, VcfHeaderConflicts) or
+        not isinstance(first, VcfHeaderConflicts)):
+      raise NotImplementedError
+    self._merge_conflicts(first.info_conflicts, second.info_conflicts)
+    self._merge_conflicts(first.format_conflicts, second.format_conflicts)
+
+  def _merge_conflicts(self, first, second):
+    # type: (Dict[str, _ConflictDetails], Dict[str, _ConflictDetails]) -> None
+    """Updates ``first`` by merging values from ``first`` and ``second``."""
+
+    for key, conflicts_details in second.iteritems():
+      if key not in first:
+        first[key] = conflicts_details
+        continue
```

Comment:
nit: since there is only one line after `continue`, consider making an `else` statement. IMO, `continue` is useful when your `else` gets very large and is complex, but makes the code a bit less readable otherwise.

Reply:
Good point.